Content.Client/Chemistry/Visualizers/SolutionContainerVisualsSystem.cs

```diff
@@ -24,6 +24,16 @@ private void OnMapInit(EntityUid uid, SolutionContainerVisualsComponent componen
 
     protected override void OnAppearanceChange(EntityUid uid, SolutionContainerVisualsComponent component, ref AppearanceChangeEvent args)
     {
+        // Check if the solution that was updated is the one set as represented
+        if (!string.IsNullOrEmpty(component.SolutionName))
+        {
+            if (AppearanceSystem.TryGetData<string>(uid, SolutionContainerVisuals.SolutionName, out var name,
+                args.Component) && name != component.SolutionName)
+            {
+                return;
+            }
+        }
+
```

Comment:
i dont fully get whats the purpose of this

Reply:
Currently SolutionContainerVisuals takes any solution inside a SolutionContainerManagerComp. I added a datafield to specify which solution you want to represent in the sprite. 
If this datafield is set, this prevents processing the appearancechanged event if another solution than the one defined is selected.

This is useful in sinks as they have two solutions : the tank, which contains clean water, and the drainBuffer which contain waste.